### PR TITLE
test: add guidepup as screen-reader automation

### LINKED ISSUE #2403: Test/include guidepup to repo
Automated screen-reader testing: https://github.com/guidepup/guidepup

-> We should test it and include it to our repo

further information in general: https://dev.to/craigmorten/a11y-testing-automating-screenreaders-1a3n

### PR DESCRIPTION
**Pull request by nmerget**
## Proposed changes

closes #2403

## Types of changes

-   [ ] Bugfix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [ ] Refactoring (fix on existing components or architectural decisions)
-   [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] Documentation Update (if none of the other choices apply)

### CODE REVIEW

File: showcases/screen-reader/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Screen Automated Reader (ScAR 🦁🔥💀)
+
+Test with this:
+
+## MacOS
+
+```shell
+npm run test:screen-reader:macos --workspace=react-showcase -- -- --ui
```
Comment (mfranzke) [inline showcases/screen-reader/README.md:8]: Why is there a duplicated `--` ? I would have only expected one of them.

File: showcases/playwright.screen-reader.windows.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+import defaultScreenReaderConfig from './playwright.screen-reader';
+
+const config: PlaywrightTestConfig = {
+	...defaultScreenReaderConfig,
+	snapshotPathTemplate:
+		'{snapshotDir}/{testFileDir}/windows/{projectName}/{testName}{ext}',
+	projects: [
+		{
+			name: 'chromium',
+			use: { ...devices['Desktop Chrome'], headless: false }
```
Comment (mfranzke) [inline showcases/playwright.screen-reader.windows.ts:11]: @nmerget the screenreader testing doesn't work headless, does it?

File: .github/actions/playwright-cache/action.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    - name: 🎄🎸🥊 Log Cache Hit
+      shell: bash
+      env:
+        HIT: ${{ steps.playwright-cache.outputs.cache-hit }}
+      run: echo $HIT
+
+    - name: ⏬ Install Playwright's dependencies
+      shell: bash
+      if: steps.playwright-cache.outputs.cache-hit != 'true'
+      run: npx playwright install --with-deps ${{ env.BROWSERS }}
```
Comment (mfranzke) [inline .github/actions/playwright-cache/action.yml:44]:
```suggestion
      run: npx --no -- playwright install --with-deps ${{ env.BROWSERS }}
```

We should use the playwright already installed and ensure that we're sticking to that for control over the playwrights version being in use.

**Review by mfranzke — APPROVED** [review]

File: showcases/playwright.screen-reader.windows.ts (hunk 2 above)
Comment (nmerget) [reply showcases/playwright.screen-reader.windows.ts:11]: nope browser has to be open

File: showcases/screen-reader/README.md (hunk 1 above)
Comment (nmerget) [reply showcases/screen-reader/README.md]: fixed

**Review by mfranzke — APPROVED** [review]

File: showcases/screen-reader/README.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Screen Automated Reader (ScAR 🦁🔥💀)
+
+Start a test with these commands:
```
Comment (mfranzke) [inline showcases/screen-reader/README.md:3]: Let's mention that the users would have to ensure a build (or watcher) previous to running the following commands, as those would only start an environment out of any previous build.

**Pull request merged**